docusaurus/docs/dev-docs/upgrade-tool.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Upgrade tool
+description: The Strapi upgrade tool is a CLI command that helps automatically upgrading to a new Strapi version.
+displayed_sidebar: devDocsSidebar
+---
+
+# Upgrade tool
+
+The upgrade tool assists Strapi users in upgrading their Strapi application dependencies and code to a specific version.
+
+Running the upgrade tool triggers the update of the application dependencies, their installation, and the execution of a series of **codemods** <Annotation>Codemods are automated tools used for large-scale code modifications, especially useful for refactoring, adapting to API changes, or applying coding style updates. They enable precise and programmable alterations to source code.</Annotation> that automatically edit the application codebase according to the breaking changes introduced up until the targeted version.
+
+The upgrade tool is a Strapi package and can be run from the CLI.
+
+## Version types
+
+Strapi version numbers respect the [semantic versioning](https://semver.org/) conventions:
+
+<ThemedImage
+  alt="Version numbers explained"
+  sources={{
+    light: '/img/assets/update-migration/version-numbers.png',
+    dark: '/img/assets/update-migration/version-numbers_DARK.png',
+  }}
+/>
+
+- The first number is the **major** version number.
+- The second number is the **minor** version number.
+- The third number is the **patch** version number.
+
+The upgrade tool allows upgrading to a major, minor, or patch version.
+
+| The latest Strapi v4 version is v4.16.2 and my Strapi application is currently on… | If I run… | My Strapi application will be upgraded to … |
+|----|----|----|
+| v4.14.1 | `npx @strapi/upgrade patch` | v4.14.6<br/><br/>(because v4.14.6 is the latest patch version for the v4.14 minor version) |
+| v4.14.1 | `npx @strapi/upgrade minor` | v4.16.2 |
+| v4.14.1 | `npx @strapi/upgrade major` | Nothing.<br/><br/>I first need to run `npx @strapi/upgrade minor` to upgrade to v4.16.2. |
+| v4.16.2 | `npx @strapi/upgrade major` | v5.0.0  |
+
+## Upgrade to a major version
+
+Run the upgrade tool with the `major` parameter to upgrade the project to the next major version of Strapi:
+
+```bash
+npx @strapi/upgrade major
+```
+
+During the upgrade process, the application dependencies are updated and installed, and the codemods up until the targeted major version are executed.
```

Comment:
since there's no "targeted major version" here, we should say "up until the next available major version" or something similar

(sorry, my Notion notes here were wrong and misleading)